php/PluginConfiguration.php

```diff
@@ -106,7 +106,22 @@ public static function setResourceGroupId(int $id): void
 
     public static function getApiProfile(): ?ProfileModel
     {
-        return get_option('coyote_api_profile', null);
+        return self::possiblyMigrateApiProfile( get_option('coyote_api_profile', null) );
+    }
+
+    /*
+     * Check if ProfileModel is outdated (v1 object)
+     * If so, retrieve v2 model and update the object in the database
+     */
+    public static function possiblyMigrateApiProfile($profile): ?ProfileModel
+    {
+        if (is_null($profile))
+            return null;
+
+        if (!$profile instanceof ProfileModel)
+            self::setApiProfile(WordPressCoyoteApiClient::getProfile());
```

Comment:
This looks like it could fail; `setApiProfile` has a signature of `ProfileModel $profile`, while `WordPressCoyoteApiClient::getProfile` returns `?ProfileModel`.